printing/printmod.ml

```diff
@@ -347,19 +346,23 @@ let rec print_typ_expr env mp locals mty =
       hov 3 (str"(" ++ (print_kn locals fapp) ++ spc () ++
 		 prlist_with_sep spc (print_modpath locals) mapp ++ str")")
   | MEwith(me,WithDef(idl,(c, _)))->
-      let env' = None in (* TODO: build a proper environment if env <> None *)
       let s = String.concat "." (List.map Id.to_string idl) in
-      (* XXX: What should env and sigma be here? *)
-      let env = Global.env () in
-      let sigma = Evd.from_env env in
-      hov 2 (print_typ_expr env' mp locals me ++ spc() ++ str "with" ++ spc()
-             ++ def "Definition"++ spc() ++ str s ++ spc() ++ str ":="++ spc()
-             ++ Printer.pr_lconstr_env env sigma c)
+      let body = match extent with
+        | WithContents ->
+            let sigma = Evd.from_env env in
+            spc() ++ str ":=" ++ spc() ++ Printer.pr_lconstr_env env sigma c
```

Comment:
Small comment: this is down to style but indeed I find that using:
```
Pp.seq [spc (); str ":=; spc (); Printer....]
```
maybe be more future-proof if we move towards a box based model.